### PR TITLE
en,zh: graceful restart tikv

### PR DESCRIPTION
**Pull request by july2993**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-tidb-operator) that's required for repo owners to accept my contribution.

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.2 (TiDB Operator 1.2 versions)
- [ ] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?
fix https://github.com/pingcap/docs-tidb-operator/issues/1483

- This PR is translated from: 
- Other reference link(s): 
https://github.com/pingcap/tidb-operator/pull/4279
### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: zh/restart-a-tidb-cluster.md [hunk 1]
```diff
@@ -50,3 +50,28 @@ spec:
     annotations:
       tidb.pingcap.com/restartedAt: 2020-04-20T12:00
 ```
+
+## Performing a graceful restart of a TiKV Pod(TiDB Operator 版本必须是 v1.2.5 或者更加新的版本)
```
Comment (ran-huang) [inline zh/restart-a-tidb-cluster.md:54]:
```suggestion
## 优雅滚动重启 TiKV Pod
```

File: zh/restart-a-tidb-cluster.md [hunk 2]
```diff
@@ -50,3 +50,28 @@ spec:
     annotations:
       tidb.pingcap.com/restartedAt: 2020-04-20T12:00
 ```
+
```
Comment (ran-huang) [inline zh/restart-a-tidb-cluster.md:53]: 这一节的优雅重启和上一节的区别是什么呢？上一节写了也可以只重启某个组件。

File: zh/restart-a-tidb-cluster.md [hunk 3]
```diff
@@ -50,3 +50,28 @@ spec:
...
+## Performing a graceful restart of a TiKV Pod(TiDB Operator 版本必须是 v1.2.5 或者更加新的版本)
+
+Operator 支持给 TiKV Pod 添加 annotation 来触发优雅重启。
```
Comment (ran-huang) [inline zh/restart-a-tidb-cluster.md:56]:
```suggestion
从 v1.2.5 起，TiDB Operator 支持给 TiKV Pod 添加 annotation 来触发优雅重启。
```

File: zh/restart-a-tidb-cluster.md (hunk 2 above)
Comment (july2993) [reply zh/restart-a-tidb-cluster.md:53]: 上面只能组建的 全部 pods (多个）, 改的对象是 tc

这个是改的 一个 Pod，只重启一个具体的  pod

Comment (ran-huang) [reply zh/restart-a-tidb-cluster.md:53]: 谢谢，明白了。能把这个信息补充到文档里吗？用户可能也有这个疑问。

Comment (july2993) [reply zh/restart-a-tidb-cluster.md:53]: 一个说的是组建全部 pod，一个说的是 kv pod, 另外从操作对象来看这个应该已经清楚了的

File: zh/restart-a-tidb-cluster.md [hunk 4]
```diff
@@ -50,3 +50,28 @@ spec:
...
+- `none`: 什么也不做。
+- `delete-pod`: 删除 Pod 并删除 PD 的 evict-leader scheduler。
+
+当值是`delete-pod`时 Operator 会做如下操作：
+
+1. 添加对应 TiKV store 的 evict-leader-scheduler。
+2. 当 leader 数掉到 0 时删除 Pod 让它重建。
+3. 当新的 Pod Ready 后删除 evict-leader-scheduler。
+
+例如你可以如下添加一个 key 为 `tidb.pingcap.com/restart` 的 annotation 来触发优雅重启：
```
Comment (ran-huang) [inline zh/restart-a-tidb-cluster.md:71]: 1. 这里是 `tidb.pingcap.com/restart`，不是 `tidb.pingcap.com/evict-leader` 吗？
2. 建议把这个 example 移到前面（L57），然后再给出 annotation key 的合法值、解释 `delete-pod` 对应的操作。这样更符合用户的阅读习惯。

File: zh/restart-a-tidb-cluster.md [hunk 5]
```diff
@@ -50,3 +50,28 @@ spec:
     annotations:
       tidb.pingcap.com/restartedAt: 2020-04-20T12:00
 ```
+
+## 优雅重启 TiKV Pod
```
Comment (DanielZhangQD) [inline zh/restart-a-tidb-cluster.md:54]:
```suggestion
## 优雅重启单个 TiKV Pod
```

File: zh/restart-a-tidb-cluster.md [hunk 6]
```diff
@@ -50,3 +50,28 @@ spec:
...
+
+从 v1.2.5 起， Operator 支持给 TiKV Pod 添加 annotation 来触发优雅重启。
```
Comment (DanielZhangQD) [inline zh/restart-a-tidb-cluster.md:56]:
```suggestion
从 v1.2.5 起， TiDB Operator 支持给 TiKV Pod 添加 annotation 来触发优雅重启。
```

File: zh/restart-a-tidb-cluster.md [hunk 7]
```diff
@@ -50,3 +50,28 @@ spec:
...
+
+Annotation key: `tidb.pingcap.com/evict-leader`
+
+Annotation 的值控制当 TiKV region leader 数掉到 0 时的行为，合法的值是以下其一：
+
+- `none`: 什么也不做。
+- `delete-pod`: 删除 Pod 并删除 PD 的 evict-leader scheduler。
```
Comment (DanielZhangQD) [inline zh/restart-a-tidb-cluster.md:63]:
```suggestion
- `delete-pod`: 删除 Pod，在 Pod 启动并且 Ready 后删除 PD 的 evict-leader scheduler。
```

File: zh/restart-a-tidb-cluster.md [hunk 8]
```diff
@@ -50,3 +50,28 @@ spec:
...
+
+当值是`delete-pod`时 Operator 会做如下操作：
```
Comment (DanielZhangQD) [inline zh/restart-a-tidb-cluster.md:65]:
```suggestion
当值是 `delete-pod` 时 TiDB Operator 会做如下操作：
```

File: zh/restart-a-tidb-cluster.md [hunk 9]
```diff
@@ -50,3 +50,28 @@ spec:
...
+1. 添加对应 TiKV store 的 evict-leader-scheduler。
+2. 当 leader 数掉到 0 时删除 Pod 让它重建。
+3. 当新的 Pod Ready 后删除 evict-leader-scheduler。
+
+例如你可以如下添加一个 key 为 `tidb.pingcap.com/restart` 的 annotation 来触发优雅重启：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl annotate pods <TiKV-pod-name> tidb.pingcap.com/evict-leader="delete-pod"
```
Comment (DanielZhangQD) [inline zh/restart-a-tidb-cluster.md:76]:
```suggestion
kubectl -n ${namespace} annotate pod ${tikv_pod_name} tidb.pingcap.com/evict-leader="delete-pod"
```

File: zh/restart-a-tidb-cluster.md [hunk 10]
```diff
@@ -50,3 +50,23 @@ spec:
...
+
+添加一个 key 为 `tidb.pingcap.com/evict-leader` 的 annotation，触发优雅重启：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl -n ${namespace} annotate pod ${tikv_pod_name} tidb.pingcap.com/evict-leader="delete-pod"
+```
+
+当 TiKV region leader 数掉到 0 时，根据 annotation 的不同值，TiDB Operator 会采取不同的行为。合法的annotation 值如下：
```
Comment (DanielZhangQD) [inline zh/restart-a-tidb-cluster.md:66]:
```suggestion
当 TiKV region leader 数掉到 0 时，根据 annotation 的不同值，TiDB Operator 会采取不同的行为。合法的 annotation 值如下：
```

File: zh/restart-a-tidb-cluster.md [hunk 11]
```diff
@@ -50,3 +50,23 @@ spec:
...
+
+- `none`: 无对应行为。
+- `delete-pod`: 删除 Pod，在 Pod 启动并且 Ready 后，删除 PD 的 evict-leader-scheduler。TiDB Operator 的具体行为如下：
```
Comment (DanielZhangQD) [inline zh/restart-a-tidb-cluster.md:69]:
```suggestion
- `delete-pod`: 删除 Pod，TiDB Operator 的具体行为如下：
```

File: zh/restart-a-tidb-cluster.md [hunk 12]
```diff
@@ -50,3 +50,23 @@ spec:
...
+    1. 调用 PD API，为对应 TiKV store 添加 evict-leader-scheduler。
+    2. 当 leader 数掉到 0 时，删除 Pod 并重建 Pod。
+    3. 当新的 Pod Ready 后，删除对应 PD 的 evict-leader-scheduler。
```
Comment (DanielZhangQD) [inline zh/restart-a-tidb-cluster.md:72]:
```suggestion
    2. 当 TiKV region leader 数掉到 0 时，删除 Pod 并重建 Pod。
    3. 当新的 Pod Ready 后，调用 PD API 删除对应 TiKV store 的 evict-leader-scheduler。
```

File: en/restart-a-tidb-cluster.md [hunk 13]
```diff
@@ -59,4 +59,25 @@ spec:
     config: {}
     annotations:
       tidb.pingcap.com/restartedAt: 2020-04-20T12:00
-```
\ No newline at end of file
+```
+
+## Perform a graceful restart to a single TiKV Pod
+
+Starting from v1.2.5, TiDB Operator supports graceful restart for a single TiKV Pod
```
Comment (DanielZhangQD) [inline en/restart-a-tidb-cluster.md:66]:
```suggestion
Starting from v1.2.5, TiDB Operator supports graceful restart for a single TiKV Pod.
```

File: en/restart-a-tidb-cluster.md [hunk 14]
```diff
@@ -59,4 +59,25 @@ spec:
...
+```
+
+When the number of TiKV region leaders drops to zero, according to the value of this annotation, TiDB Operator might have different behaviors:
+
+- `none`: TiDB Operator does nothing.
+- `delete-pod`: TiDB Operator deletes the Pod by taking the following steps:
+
+    1. TiDB Operator calls the PD API and adds evict-leader-scheduler for the TiKV store.
+    2. When the number of TiKV region leaders drops to zero, TiDB Operator deletes the Pod and recreates it.
+    3. When the new Pod becomes ready, remove evict-leader-scheduler from the TiKV store by calling the PD API.
```
Comment (DanielZhangQD) [inline en/restart-a-tidb-cluster.md:83]:
```suggestion
    3. When the new Pod becomes ready, remove the evict-leader-scheduler for the TiKV store by calling the PD API.
```

**Review by ran-huang — APPROVED** [review]

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Pull request merged**